Search: Fix search pricing request before site is connected

kangzj opened the pull request:
#### Changes proposed in this Pull Request:
The PR fixes a regression introduced by #24795. `get_pricing_from_wpcom` would be called before site is connected, meaning we wouldn't be able to `wpcom_json_api_request_as_blog`. The PR also added `my-jetpack` as dependency for the `search` package.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Go to a fresh created JN site (or ensure blog is not connected, blog token is empty and the blog doesn't have search subscription)
* Ensure search pricing is correct on page `/wp-admin/admin.php?page=jetpack-search`
* Connect the site
* Ensure search pricing is correct in My Jetpack `/wp-admin/admin.php?page=my-jetpack#/add-search`

kangzj committed:
use plain wp_remote_get as connection might not be ready

kangzj committed:
changelog

kangzj committed:
update lock file

leogermani committed:
add memoization

leogermani reviewed (changes requested):
It tests well for me. I've just add memoization to the method to avoid it making to requests to wpcom on one request.

Then I realized that the Search package is using it but is not explicitly addin My Jetpack as a dependency. I think we need to do that.

kangzj commented on projects/packages/my-jetpack/src/products/class-search.php:
```diff
@@ -135,19 +136,24 @@ public static function get_wpcom_product_slug() {
-			array( 'timeout' => 2 ),
-			null,
-			'wpcom'
+		static $pricing = null;
```
It's unusual but the pricing changes with record count, so probably we'll need to take that in account? @leogermani

kangzj committed:
add my jetpack as dependecy for search

kangzj committed:
changelog

kangzj committed:
Memoize with parameter considered

kangzj committed:
Merge branch 'trunk' into fix/use-plain-remote-request-for-pricing

kangzj committed:
remove test code

kangzj commented:
Thanks for the review @leogermani 

>  I've just add memoization to the method to avoid it making to requests to wpcom on one request.

Good catch! Added memoize with parameter to avoid cached return doesn't reflect the record_count parameter.

> Then I realized that the Search package is using it but is not explicitly addin My Jetpack as a dependency. I think we need to do that.

Good catch. Added thanks!

kangzj committed:
lock file update
